### PR TITLE
Release fix: Prestart assumptions about environment do not hold anymore in kubecf/operator.

### PR DESCRIPTION
**Pull request by andreas-kupries**
Ref:
  - https://jira.suse.com/browse/CAP-520
  - https://github.com/SUSE/kubecf/issues/31

Define the unbound variables with proper ERB substitutions referencing job properties.

### CODE REVIEW

File: jobs/acceptance-tests-brain/templates/pre-start.erb [hunk 1]
```diff
@@ -4,7 +4,9 @@ set -o errexit
 set -o nounset
 
 PATH=$PATH:/var/vcap/packages/cli/bin
+API="<%= properties.smoke_tests.api %>"
+PASSWORD="<%= properties.acceptance_tests_brain.password %>"
 
-cf api --skip-ssl-validation "https://api.$DOMAIN"
-cf auth admin "${CLUSTER_ADMIN_PASSWORD}"
+cf api --skip-ssl-validation "${API}"
```
Comment (mook-as) [inline jobs/acceptance-tests-brain/templates/pre-start.erb]: Looking at [`properties.smoke_tests.api`](https://github.com/SUSE/brain-tests-release/blob/fe15bb7fa6e9f681644dbb02517296ae9af0121e/jobs/acceptance-tests-brain/spec#L45) it doesn't contain the scheme (`https://`); should we not leave that part in?  Or use [`acceptance_tests_brain.domain`](https://github.com/SUSE/brain-tests-release/blob/fe15bb7fa6e9f681644dbb02517296ae9af0121e/jobs/acceptance-tests-brain/spec#L23) instead?

Also, I don't think we actually need temporary variables here:
```suggestion
cf api --skip-ssl-validation https://api.<%= acceptance_tests_brain.domain.shellescape %>
```

Yeah, we might need to do shell escape, especially for that password…

File: jobs/acceptance-tests-brain/templates/pre-start.erb [hunk 2]
```diff
@@ -4,7 +4,9 @@ set -o errexit
...
+cf auth admin "${PASSWORD}"
```
Comment (mook-as) [inline jobs/acceptance-tests-brain/templates/pre-start.erb]: It looks like there's a [`acceptance_tests_brain.user`](https://github.com/SUSE/brain-tests-release/blob/fe15bb7fa6e9f681644dbb02517296ae9af0121e/jobs/acceptance-tests-brain/spec#L31) that we might want to use.

File: jobs/acceptance-tests-brain/templates/pre-start.erb (hunk 1 above)
Comment (andreas-kupries) [reply jobs/acceptance-tests-brain/templates/pre-start.erb]: In the brain test instance group I have

```
        smoke_tests:
          api:                 https://api.((system_domain))
```

... Hm. I made this different from the scf property. Ok. I will change the property in the manifest to match what we had in scf, i.e. without scheme, and then add it back in the release.

File: jobs/acceptance-tests-brain/templates/pre-start.erb (hunk 2 above)
Comment (andreas-kupries) [reply jobs/acceptance-tests-brain/templates/pre-start.erb]: True. Will do.

File: jobs/acceptance-tests-brain/templates/pre-start.erb (hunk 1 above)
Comment (andreas-kupries) [reply jobs/acceptance-tests-brain/templates/pre-start.erb]: Addressed with commit 39e2b8112e304c5d8b3310bb13cb94fce61c83e7.

File: jobs/acceptance-tests-brain/templates/pre-start.erb (hunk 2 above)
Comment (andreas-kupries) [reply jobs/acceptance-tests-brain/templates/pre-start.erb]: Addressed with commit 39e2b8112e304c5d8b3310bb13cb94fce61c83e7.

File: jobs/acceptance-tests-brain/templates/pre-start.erb [hunk 3]
```diff
@@ -5,6 +5,9 @@ set -o nounset
 
 PATH=$PATH:/var/vcap/packages/cli/bin
 
-cf api --skip-ssl-validation "https://api.$DOMAIN"
-cf auth admin "${CLUSTER_ADMIN_PASSWORD}"
+cf api --skip-ssl-validation \
+   "https://<%= properties.smoke_tests.api.shellescape %>"
```
Comment (mook-as) [inline jobs/acceptance-tests-brain/templates/pre-start.erb]: I think we need to `<% require 'shellwords' %>` at the top of the file? (Of course, if this actually works then that's fine.)

Also, do we use [`acceptance_tests_brain.domain`](https://github.com/SUSE/brain-tests-release/blob/fe15bb7fa6e9f681644dbb02517296ae9af0121e/jobs/acceptance-tests-brain/spec#L23) anywhere?

Also, please unquote things that have already been `shellescape`d; that would result in double-quoting.  Sometimes.

Comment (andreas-kupries) [reply jobs/acceptance-tests-brain/templates/pre-start.erb]: The `acceptance_tests_brain.domain` is translated into `CF_DOMAIN`, which in turn is used in six tests, as well as the `testutils.rb` used by all the tests.

Will make changes. Will see if I can test this locally before making the next commit.

Comment (andreas-kupries) [reply jobs/acceptance-tests-brain/templates/pre-start.erb]: Addressed with commit c8efc353b20a4b368a2a0a509ce4fbf92eff7b03.

**Comment by andreas-kupries** [comment]
@mook-as @f0rmiga This PR is now ready to merge.
It now also fixes the missing `ruby-2.3` package.

Brain tests are starting, running, completing.
While we have 10 :boom: we also have 6 :heavy_check_mark: (and 3 skipped).

**Pull request merged**